src/core/server/http/router/router.ts

```diff
@@ -312,8 +312,8 @@ export class Router implements IRouter {
   }
 }
 
-const isDataSourceConfigError = (error: any) => {
-  return error.constructor.name === 'DataSourceConfigError' && error.statusCode === 400;
+const isDataSourceError = (error: any) => {
+  return error.constructor.name === 'DataSourceError' && error.statusCode === 400;
```

Comment:
can we use `instanceof`instead of `error.constructor.name` ?

Reply:
importing `DataSourceError` from plugin module to here(core module) doesn't sound like a good practice. But anyway, I did some refactor to move data source error response logic from `router.ts` to `resolve_index.ts`(/internal/index-pattern-manage/resolve-index).  

I checked the codebase, only the route handler defined  resolve_index is missing error handling. All other `internal/xxx/xxx` API has err handling. E.g `internal/search/`, `/internal/_msearch`, `/internal/index-pattern-manage/preview-script`

    ```
    catch (err) {
            return res.customError({
              statusCode: err.statusCode,
              body: {
                message: err.message,
                attributes: {
                  error: err.body.error,
                },
              },
            });
    ```   

@zengyan-amazon could you review again